src/main/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,41 @@
+package study;
+
+public class StringAddCalculator {
+
+    private static final int NULL_EMPTY_NUM = 0;
+    public static int splitAndSum(String text) {
+        if (text == null || text.isEmpty()){
+            return NULL_EMPTY_NUM;
+        }
+
+        String[] stringList = text.split("[^0-9]");
+
+        return sum(stringList);
+    }
+
+    private static int sum(String[] stringList) {
```

Comment:
> 분명.. 어제 저녁에 리팩토링 하는 방법을 배웠는데 직접 해보려니 쉽지 않네요 ㅠㅠ.. 더 할 수 있을 것 같은데 막상 함수로 바꿔보니 return을 어떻게 처리해야할지 찾지 못했습니다.

리팩토링이 많이 어려운 부분이죠 ㅠㅠ
리팩토링을 해야하는 신호는 크게 다음과 같다고 생각하는데요!

1. 테스트를 못하는 구조는 리팩토링이 필요한 신호
2. 메서드가 2가지 이상의 기능을 수행하고 있을 때
3. 메서드가 하고자하는 기능과 파라미터가 모순이 있을 때

현재 잘해주셔서 테스트가 가능하지만 메서드들이 2가지 이상의 일을 수행하고 있으며
기능과 파라미터의 불일치 모순이 조금 보이는 것 같아요 👀 

가령 sum()을 해준다는건 숫자들의 합을 구해줘야하는데 파라미터가 String 배열을 받음으로써
정수로 parsing 해주는 기능이 추가되는 구조 같아요..!
메서드를 단순하게 만드는 방향으로 리팩토링을 진행해보시면 어떨까요 :) ?

```suggestion
    private static int sum(int[] numbers) {
```